src/init/features/dataconnect/sdk.ts

```diff
@@ -224,10 +224,17 @@ export async function generateSdkYaml(
   return connectorYaml;
 }
 
-export async function actuate(sdkInfo: SDKInfo) {
+export async function actuate(sdkInfo: SDKInfo, config?: Config) {
```

Comment:
Is there a reason that this is optional (and we have the fallback behavior)? Look like actuate is only ever called in one place, and config can't be undefined there, so it feels like we can simplify to just:

```
    logBullet(`Writing your new SDK configuration to ${connectorYamlPath}`);
    await config.askWriteProjectFile(
      path.relative(config.projectDir, connectorYamlPath),
      sdkInfo.connectorYamlContents,
    );
```

Reply:
Ah, the reason for that was that we call `actuate` from the tests, and don't have a `Config` to pass to `actuate`

Reply:
Alternatively, I was thinking of doing something like this:
```
export async function actuate(sdkInfo: SDKInfo, config: Config, skipCheck?: boolean) {
  const connectorYamlPath = `${sdkInfo.connectorInfo.directory}/connector.yaml`;
  logBullet(`Writing your new SDK configuration to ${connectorYamlPath}`);
  if(skipCheck) {
    await fs.writeFile(connectorYamlPath, sdkInfo.connectorYamlContents);
  } else {
    await config.askWriteProjectFile(
      path.relative(config.projectDir, connectorYamlPath),
      sdkInfo.connectorYamlContents,
    );
  }
  ```

Reply:
It's bad practice to have branch behavior for tests - then, we're testing something different than what were actually running. You can just make a config in the test like so:
```
new Config({}, { projectDir: "test", cwd: "test" }),
```
You'll also want to stub out askWriteProjectFile in the test (see https://github.com/firebase/firebase-tools/blob/master/src/init/features/functions.spec.ts#L80 for an example)

Reply:
That's fair, I should do this the *right* way then :-)